### PR TITLE
fix issue #3252

### PR DESCRIPTION
**Pull request by zhaoshengjun**
**Please check if the PR fulfills these requirements**
- [x ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
- [ x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

**What is the current behavior?** (You can also link to an open issue here)
Default argument assignment is missing. Ref #3252 

**What is the new behavior?**
Default argument assignment will be correctly placed.

**Does this PR introduce a breaking change?**
- [ ] Yes
- [x ] No

If this PR contains a breaking change, please describe the following... 

* Impact:
* Migration path for existing applications: 
* Github Issue(s) this is regarding:

**Other information**:

### CODE REVIEW

**Comment by zhaoshengjun** [comment]
Test passed on `node 7`, but failed on `node 6` due to `yarn`.

File: test/helpers/supportDefaultAssignment.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+module.exports = function supportDefaultAssignment() {
+	try {
+		return true;
```
Comment (sokra) [inline test/helpers/supportDefaultAssignment.js]: This doesn't look correct.

File: lib/Parser.js [hunk 2]
```diff
@@ -923,6 +923,8 @@ Parser.prototype.enterObjectPattern = function enterObjectPattern(pattern, onIde
 			case "AssignmentProperty":
 				this.enterPattern(property.value, onIdent);
 				break;
+			default:
+				this.enterPattern(property.value, onIdent);
 		}
```
Comment (sokra) [inline lib/Parser.js]: Could you remove the whole switch-case and shorten this to `this.enterPattern(property.value, onIdent)`?

From AST explorer it looks like `property.type` is always `Property`.

File: test/helpers/supportDefaultAssignment.js (hunk 1 above)
Comment (zhaoshengjun) [inline test/helpers/supportDefaultAssignment.js]: I just want it to be filtered.

File: lib/Parser.js (hunk 2 above)
Comment (zhaoshengjun) [inline lib/Parser.js]: Sure, I will make the change.

File: test/helpers/supportDefaultAssignment.js (hunk 1 above)
Comment (sokra) [inline test/helpers/supportDefaultAssignment.js]: Check how the other helpers work:

https://github.com/webpack/webpack/blob/master/test/helpers/supportsDefaultArgs.js
https://github.com/webpack/webpack/blob/master/test/helpers/supportsBlockScoping.js
https://github.com/webpack/webpack/blob/master/test/helpers/supportsES6.js

**Comment by sokra** [comment]
Thanks

**Comment by zhaoshengjun** [comment]
My pleasure!
